### DISCUSSION TITLE
How to adjust the width of a table to the content when you knit to HTML?

### DISCUSSION THREAD

**Discussion by andresrcs** [quarto-dev/quarto-cli#3912] [Q&A]
This is probably a simple one but the correct search terms are eluding me, how to set the column width to adjust to content for HTML tables in quarto? specifically, `gt` tables.

In markdown, this was the default behavior but with quarto the table maximizes to take the full space, consider this simple example where you have a very small table.

````
---
title: "Test"
format:
  html:
    self-contained: true
knitr:
  opts_chunk: 
    echo: false
    warning: false
    message: false
editor: visual
---

```{r}
#| tbl-colwidths: false

library(dplyr)
library(gt)

iris %>% 
    count(Species) %>% 
    gt()
```
````

When executed interactively the table renders small on the RStudio's Visual Editor, but when rendered, the table expands to the full page layout width.

![image](https://user-images.githubusercontent.com/14981085/211439598-f6dda532-48de-44ad-a435-4ff028103673.png)
![image (1)](https://user-images.githubusercontent.com/14981085/211439616-a8dbc23b-eb85-4190-a916-e6fd3fcfdb36.png)

**Comment by mcanouil** [comment]
Add `as_raw_html()` to avoid Quarto modifying the HTML table. 😉

```r
iris %>% 
  count(Species) %>% 
  gt() %>%
  as_raw_html()
```

**Reply by cscheid** [reply]
That's a good workaround.

To expand a little on that answer: quarto currently does modify the output of a `gt` table to better fit its own themes. @andresrcs as you can tell, the results are not always great. We're currently prototyping a more general system for quarto and downstream libraries to communicate about styles and postprocessing. We're confident that in the next few months we'll have a system that allows for more explicit control, without having to resort to these workarounds.

**Reply by andresrcs** [reply]
Thanks for the workaround. I was hoping this was just an obscure option I couldn't find, at least now I can stop looking and wait for the new feature.
